hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreFlusher.java

```diff
@@ -924,4 +947,62 @@ public boolean equals(Object obj) {
       return compareTo(other) == 0;
     }
   }
+
+  private int getHandlerCount(Configuration conf) {
+    int handlerCount = conf.getInt("hbase.hstore.flusher.count", 2);
+    if (handlerCount < 1) {
+      LOG.warn(
+        "hbase.hstore.flusher.count was configed to {} which is less than 1, " + "corrected to 1",
+        handlerCount);
+      handlerCount = 1;
+    }
+    return handlerCount;
+  }
+
+  @Override
+  public void onConfigurationChange(Configuration newConf) {
+    int newHandlerCount = getHandlerCount(newConf);
+    if (newHandlerCount != flushHandlers.length) {
+      LOG.info("update hbase.hstore.flusher.count from {} to {}", flushHandlers.length,
+        newHandlerCount);
+      lock.writeLock().lock();
+      try {
+        FlushHandler[] newFlushHandlers = new FlushHandler[newHandlerCount];
+        if (newHandlerCount > flushHandlers.length) {
+          System.arraycopy(flushHandlers, 0, newFlushHandlers, 0, flushHandlers.length);
```

Comment:
Use Arrays.copyOf, it can handle both cases, where the new length is greater or less.

Reply:
Thanks for your suggestion. Will address this in the next commit. Still working on fixing the UT.